MatrixSDK/VoIP/MXCallManager.m

```diff
@@ -565,6 +632,44 @@ - (void)handleCallRejectReplacement:(MXEvent *)event
     }
 }
 
+- (void)handleRoomMember:(MXEvent *)event
+{
+    MXRoomMemberEventContent *content = [MXRoomMemberEventContent modelFromJSON:event.content];
+    
+    if ([MXTools membership:content.membership] == MXMembershipInvite)
+    {
+        //  a room invite
+        MXRoom *room = [_mxSession roomWithRoomId:event.roomId];
+        NSString *directUserId = room.directUserId;
+        
+        if (!directUserId && content.isDirect)
+        {
+            directUserId = event.sender;
+        }
+        
+        if (directUserId)
+        {
+            [self getNativeUserFrom:directUserId success:^(MXThirdPartyUserInstance * _Nonnull user) {
+                MXRoom *nativeRoom = [self.mxSession directJoinedRoomWithUserId:user.userId];
+                if (nativeRoom)
+                {
+                    //  auto-accept this invite from the virtual room, if a direct room with the native user found
+                    [room join:^{
+                        NSLog(@"[MXCallManager] handleRoomMember: auto-joined on virtual room successfully.");
+                        
+                        //  set account data on the room, if required
+                        [self.mxSession.roomAccountDataUpdateDelegate updateAccountDataIfRequiredForRoom:room
```

Comment:
does it worth to do it here?
The info will come back to the `handleSyncResponse` as well.

Reply:
Actually it won't, we're only joining a room here. We will lose the virtual room if we don't mark it here.

And even it would, it's better to do it earlier, otherwise we would need to traverse all the rooms to find the related virtual room, because we have only connection from the virtual room to the real room, not vice versa.